src/components/shared/globalTypes.ts

```diff
@@ -195,7 +211,31 @@ export class Directory extends FileSystemObject {
     return currentFsObject;
   }
 
+  isEmpty(): boolean {
+    return this.children === undefined || this.children.size === 0;
```

Comment:
Which language are you using? `.size`?

Reply:
hahaha, thank you for catching this! But this raises a good pt about why unit tests should probably be made. It would've caught this error.